lib/src/app/provisioning.dart

```diff
@@ -0,0 +1,52 @@
+import 'dart:async';
+
+import '../gen/provisioning/v1/provisioning.pbgrpc.dart';
+
+enum NetworkType { wifi, wired }
+
+/// gRPC client for connecting to Viam's Provisioning Service
+///
+/// All calls must be authenticated.
+class ProvisioningClient {
+  final ProvisioningServiceClient _client;
+
+  ProvisioningClient(this._client);
+
+  /// Get the status of the SmartMachine
+  Future<GetSmartMachineStatusResponse> getSmartMachineStatus() async {
+    final request = GetSmartMachineStatusRequest();
+    final response = await _client.getSmartMachineStatus(request);
+    return response;
+  }
+
+  /// Set the network credentials of the Smart machine, so it can connect to the internet.
+  ///
+  /// [ssid] is the networks ssid
+  /// [psk] is the networks passkey
+  Future<void> setNetworkCredentials({NetworkType type = NetworkType.wifi, required String ssid, required String psk}) async {
```

Comment:
`ssid` and `psk` shouldn't be required for `wired` connections. Not a locker, but please add a TODO